posix/select.c

```diff
@@ -0,0 +1,224 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <openenclave/corelibc/stdio.h>
+#include <openenclave/corelibc/stdlib.h>
+#include <openenclave/corelibc/string.h>
+#include <openenclave/corelibc/sys/epoll.h>
+#include <openenclave/corelibc/sys/select.h>
+#include <openenclave/corelibc/time.h>
+#include <openenclave/internal/posix/raise.h>
+#include <openenclave/internal/print.h>
+#include <openenclave/internal/trace.h>
+
+void _set_to_fd_list(
+    oe_fd_set* set,
+    int flags,
+    int fdlist_max,
+    int* fd_list,
+    int* fd_flags)
+{
+    int fd_idx = 0;
+    uint32_t idx = 0;
+    uint32_t* pbits = (uint32_t*)set->fds_bits;
+    uint32_t bitpos;
+
+    for (idx = 0; idx<OE_FD_SETSIZE>> 5; idx++)
+    {
+        uint32_t bitmask = 1;
+        for (bitpos = 0; bitpos < 8 * sizeof(long); bitpos++)
+        {
+            if (pbits[idx] & bitmask)
+            {
+                uint32_t fd = (idx << 5) + bitpos;
+                if (fd_idx < fdlist_max)
+                {
+                    for (fd_idx = 0; fd_idx < fdlist_max; fd_idx++)
+                    {
+                        if ((fd_list[fd_idx] == (int)fd) ||
+                            (fd_list[fd_idx] == (int)0xffffffff))
```

Comment:
Maybe fd_list should be initialized to 0xffffffff somewhere and perhaps it was, I cant find where though.